docs/dev-notes/BACKGROUND_QUEUE.md

```diff
@@ -0,0 +1,3 @@
+# Background queue
+
+See [the iOS SDK doc](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/BACKGROUND-QUEUE.md) to maintain a single source of truth of this knowledge. 
```

Comment:
The .md docs in the Android and iOS SDK are currently *not* hosted on github pages. 

Are you suggesting instead of having a URL that is hard-coded to a git branch like this is:
```
https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/BACKGROUND-QUEUE.md
                                             ^^^^^^^^^^^^
```

We have a URL for a webpage that has a higher chance of not being a 404?

Reply:
> also, redirecting to ios repo doesn't give the best experience. As it might suggest that information might be spread among different repos.

I do agree. I felt that this solution of linking to the iOS docs was a better idea then copying and pasting the identical document into this repo. 

I did think about making a new github repo just for storing SDK docs but felt that was overkill. I did think of an internal doc in notion but I felt having the doc live in the repo (1) helps us keep it more up-to-date and (2) it's a document that could be helpful for customers so it's good to live here. 

The solution I have here in the PR is what I found to be the best idea that I could come up with. If you have a better idea, I would love to hear it.

Reply:
>I did think about making a new github repo just for storing SDK docs but felt that was overkill. I did think of an internal doc in notion but I felt having the doc live in the repo (1) helps us keep it more up-to-date and (2) it's a document that could be helpful for customers so it's good to live here.

Maybe not in the PR but we might have to move to one of these approaches, we already have a lot of information regarding the development environment, CI/CD pipelines, BQ and this list is going to grow. So either a separate repo or a notion would do. 

This can be closed and maybe a ticket can be created out of this.